cmd/version_check.go

```diff
@@ -0,0 +1,270 @@
+// Copyright © 2017 NAME HERE <EMAIL ADDRESS>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/dcos/dcos-checks/client"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+// NewVersionCheck returns an initialized instance of *ComponentCheck.
+func NewVersionCheck(name string) DCOSChecker {
+	check := &VersionCheck{Name: name}
+	check.ClusterLeader = "leader.mesos"
+	return check
+}
+
+// MasterListResponse response for leader.mesos/master.mesos
+type MasterListResponse []struct {
+	Host string `json:"host"`
+	IP   string `json:"ip"`
+}
+
+// AgentListResponse response for /slaves
+type AgentListResponse struct {
+	Slaves []struct {
+		ID         string `json:"id"`
+		Hostname   string `json:"hostname"`
+		Port       int    `json:"port"`
+		Attributes struct {
+			PublicIP string `json:"public_ip"`
+		} `json:"attributes"`
+	} `json:"slaves"`
+	RecoveredSlaves []interface{} `json:"recovered_slaves"`
+}
+
+// VersionResponse responses /dcos-metadata/dcos-version.json
+type VersionResponse struct {
+	Version         string `json:"version"`
+	DcosImageCommit string `json:"dcos-image-commit"`
+	BootstrapID     string `json:"bootstrap-id"`
+}
+
+// URLFields is used to construct the url
+type URLFields struct {
```

Comment:
I am not sure what you mean, this is so that we can pass around whats needed to generate a url. I want the getURL function to move to checks.go